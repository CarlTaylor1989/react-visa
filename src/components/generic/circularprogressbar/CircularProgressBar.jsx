import React from 'react';
import PropTypes from 'prop-types';

import './CircularProgressBar.scss';

const CircularProgressBar = ({
  size, // Size of the enclosing square
  strokeWidth,
  percentage
}) => {
  // SVG centers the stroke width on the radius, subtract out so circle fits in square
  const radius = (size - strokeWidth) / 2;
  // Enclose cicle in a circumscribing square
  const viewBox = `0 0 ${size} ${size}`;
  // Arc length at 100% coverage is the circle circumference
  const dashArray = radius * Math.PI * 2;
  // Scale 100% coverage overlay with the actual percent
  const dashOffset = -(dashArray - dashArray * percentage / 100);

  return (
    <svg
      className="circularProgressBar"
      width={size}
      height={size}
      viewBox={viewBox}
    >
      <circle
        className="circle-background"
        cx={size / 2}
        cy={size / 2}
        r={radius}
        strokeWidth={`${strokeWidth}px`}
      />
      <circle
        className="circle-progress"
        cx={size / 2}
        cy={size / 2}
        r={radius}
        strokeWidth={`${strokeWidth}px`}
        // Start progress marker at 12 O'Clock
        transform={`rotate(-90 ${size / 2} ${size / 2})`}
        style={{
          strokeDasharray: dashArray,
          strokeDashoffset: dashOffset
        }}
      />
    </svg>
  );
};

CircularProgressBar.defaultProps = {
  percentage: 100,
  size: 134,
  strokeWidth: 8
};

CircularProgressBar.propTypes = {
  percentage: PropTypes.number,
  size: PropTypes.number,
  strokeWidth: PropTypes.number
};

export default CircularProgressBar;
