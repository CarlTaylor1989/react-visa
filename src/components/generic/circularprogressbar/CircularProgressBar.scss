@import '../../../styles/variables';

.circularProgressBar {
  .circle-background,
  .circle-progress {
    fill: none;
  }

  .circle-background {
    stroke: none;
  }

  .circle-progress {
    stroke: $colour-yellow;
    stroke-linecap: round;
    stroke-linejoin: round;
  }
}
