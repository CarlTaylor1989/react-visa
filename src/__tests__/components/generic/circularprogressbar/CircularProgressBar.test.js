/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import CircularProgressBar from
  '../../../../components/generic/circularprogressbar/CircularProgressBar';

test('renders the CircularProgressBar component', () => {
  const wrapper = shallow(<CircularProgressBar />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
