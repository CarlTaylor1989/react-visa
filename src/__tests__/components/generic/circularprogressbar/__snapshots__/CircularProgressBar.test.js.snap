// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CircularProgressBar component 1`] = `
<svg
  className="circularProgressBar"
  height={134}
  viewBox="0 0 134 134"
  width={134}
>
  <circle
    className="circle-background"
    cx={67}
    cy={67}
    r={63}
    strokeWidth="8px"
  />
  <circle
    className="circle-progress"
    cx={67}
    cy={67}
    r={63}
    strokeWidth="8px"
    style={
      Object {
        "strokeDasharray": 395.84067435231395,
        "strokeDashoffset": -0,
      }
    }
    transform="rotate(-90 67 67)"
  />
</svg>
`;
